tla-bmcmt/src/main/scala/at/forsyte/apalache/tla/bmcmt/arena/ArenaContextSync.scala

```diff
@@ -0,0 +1,373 @@
+package at.forsyte.apalache.tla.bmcmt.arena
+
+import at.forsyte.apalache.tla.bmcmt.ArenaCell
+import at.forsyte.apalache.tla.bmcmt.smt.SolverContext
+import at.forsyte.apalache.tla.bmcmt.types.{CellT, CellTFrom, InfSetT}
+import at.forsyte.apalache.tla.lir._
+import at.forsyte.apalache.tla.types.tla
+
+/**
+ * Declares the same interface as [[Arena]]s (with the exception of [[ElemPtr]] usage).
+ *
+ * As a transitional step, this class serves to replace [[Arena]]s, but internally separately delegates code to either
+ * the context or the arena. In the future, rules will handle context operations and this class will be retired in favor
+ * of [[PureArena]]s.
```

Comment:
As far as I understand, `ArenaContextSync` serves as a temporary adapter from `Arena` to `PureArena`. Shall we call it `PureArenaAdapter`?

Reply:
Sure